### DISCUSSION TITLE
Modules not being parsed in HLS

### DISCUSSION THREAD

**Discussion by billksun** [srid/haskell-template#108] [Q&A] [answered]
Commit 27aacd1 seem to have caused issues when modules/source files are added inside of `src`. For example, when I add `src/Lib/Lib.hs`, HLS cannot parse the file and I get this error:

```
Failed to run ["cabal","v2-repl","/home/bill/Projects/test/src/Lib/Lib.hs"] in directory "/home/bill/Projects/sari". Consult the logs for full command and error.
Failed command: cabal --builddir=/home/bill/.cache/hie-bios/dist-test-46b5963281c285ba43873006efe139c6 v2-repl --with-compiler /home/bill/.cache/hie-bios/wrapper-b54f81dea4c0e6d1626911c526bc4e36 --with-hc-pkg /home/bill/.cache/hie-bios/ghc-pkg-1267337f2438d8be4f64a47aa79ad3bf /home/bill/Projects/test/src/Lib/Lib.hs
Error: cabal: Failed extracting script block: `{- cabal:` start marker not
found
Process Environment:
HIE_BIOS_GHC: /nix/store/4wjl91hrizxghwqy18a1337gq2y9mh40-ghc-9.2.7/lib/ghc-9.2.7/bin/ghc
HIE_BIOS_GHC_ARGS: -B/nix/store/73mpvqn6z1sh3y1mx9g716wmfv1qmfqv-ghc-9.2.7-with-packages/lib/ghc-9.2.7
cradle
```

To get rid of this error, I had to manually add back the  `component` and `path` parameters in `hie.yaml`:
```
cradle:
  cabal:
    - component: exe:test
      path: src

**Comment by srid** [comment, marked as answer]
Did you update the cabal file? viz.:

```diff
diff --git a/haskell-template.cabal b/haskell-template.cabal
index 770a028..3af415f 100644
--- a/haskell-template.cabal
+++ b/haskell-template.cabal
@@ -94,3 +94,5 @@ common shared
 executable haskell-template
   import:  shared
   main-is: Main.hs
+  other-modules:
+    Lib.Lib
diff --git a/src/Main.hs b/src/Main.hs
index fc5ea8d..0295d72 100644
--- a/src/Main.hs
+++ b/src/Main.hs
@@ -1,6 +1,7 @@
 module Main where
 
 import Main.Utf8 qualified as Utf8
+import Lib.Lib
 
 {- |
  Main entry point.
@@ -12,3 +13,4 @@ main = do
   -- For withUtf8, see https://serokell.io/blog/haskell-with-utf8
   Utf8.withUtf8 $ do
     putTextLn "Hello 🌎"
+    print foo

```

**Reply by billksun** [reply]
The missing `other-modules` in the cabal file, indeed is the issue here.

Hmm, sometimes the behavioral discrepancy between GHC and Cabal is a real head scratcher. GHC had no problems compiling the project (actually `cabal build` works just fine), but HLS, which uses Cabal, just can't find the modules.

**Reply by SkyLeite** [reply]
Ran into this same issue. Would it be possible to update Cabal automatically somehow to include these modules? It's a pretty confusing error for a newcomer, and it took me a while to get here :p

**Reply by srid** [reply]
@SkyLeite Yea, you can use `hpack`, in addition to using [`pre-commit-hooks.nix`](https://flake.parts/options/pre-commit-hooks-nix.html) (its hpack hook) to make sure that the cabal files are in kept in sync. Here's an example: https://github.com/nammayatri/nammayatri/tree/main/Backend

**Reply by SkyLeite** [reply]
Thank you!
